### PR TITLE
Angular task part 1

### PR DESCRIPTION
**Pull request by dzmitrybek**
- implement getting new from NewsApi
- implement news searching and loading more news
- create a section for  mock user's news
- create add/edit form for news with mock saving functionality
- add routing

### CODE REVIEW

File: angular-part/src/app/core/models/news.model.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface NewsModel {
+    id?: number;
+    source: {
```
Comment (slowpokex) [inline angular-part/src/app/core/models/news.model.ts:3]: Please extract to SourceModel interface

Comment (slowpokex) [reply angular-part/src/app/core/models/news.model.ts:3]: Nested interface isn't good i guess

File: angular-part/src/app/core/services/news.service.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        { params: { q: searchText } })
+      .pipe(
+        map((data: any) => {
+          return data.articles ? data.articles.map(this.mapToNewsItem) : [];
+        })
+      );
+  }
+
+  public convertFormDataToNews(data): NewsModel {
+    return {
```
Comment (slowpokex) [inline angular-part/src/app/core/services/news.service.ts:61]: On my side I would to extract all converters and mappers as below into static methods in NewsModel and make NewsModel as class

**Review by slowpokex — APPROVED** [review]
Good job! You really mastered in this

File: angular-part/src/app/core/services/news.service.ts (hunk 2 above)
Comment (dzmitrybek) [reply angular-part/src/app/core/services/news.service.ts:61]: I agree with you, it is a good point so I will fix it in the next part of the task

**Comment by dzmitrybek** [comment]
dfdsf
<pre>

asds
some <b>textfds</b> sdf
sdfdsf
sdf

</pre>

**Pull request merged**